test/evp_extra_test.c

```diff
@@ -4622,11 +4623,15 @@ static int test_ecx_short_keys(int tst)
     EVP_PKEY *pkey;
 
 
+    if (testctx != NULL)
+        return TEST_skip("ecx keys not supported in null provider");
```

Comment:
The message should be: `test does not work with non-default context`

Hmm, you could actually use EVP_PKEY_new_raw_private_key_ex() function instead and pass the testctx there and the TEST_skip would not be needed then. I am sorry for not realizing this earlier.

Reply:
agreed, that's even better!